# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1302605335 -7200
#      Tue Apr 12 12:48:55 2011 +0200
# Node ID 12697e9bb608df604e5218c1a182e57a79911fa3
# Parent  1e76b221bc4aa2c4b5c26a094d9804cfe1dddd09
Files can be edited and saved.

diff --git a/gui//src/FileEditorMdiSubWindow.cpp b/gui//src/FileEditorMdiSubWindow.cpp
--- a/gui//src/FileEditorMdiSubWindow.cpp
+++ b/gui//src/FileEditorMdiSubWindow.cpp
@@ -1,28 +1,38 @@
 #include "FileEditorMdiSubWindow.h"
 #include <QVBoxLayout>
 #include <QApplication>
 #include <QFile>
+#include <QFileDialog>
 
 FileEditorMdiSubWindow::FileEditorMdiSubWindow(QWidget *parent)
     : QMdiSubWindow(parent) {
     construct();
 }
 
 void FileEditorMdiSubWindow::loadFile(QString fileName) {
+    m_fileName = fileName;
     setWindowTitle(fileName);
     QFile file(fileName);
     file.open(QFile::ReadOnly);
 
     m_codeEdit->setPlainText(file.readAll());
 
     file.close();
 }
 
+void FileEditorMdiSubWindow::saveFile() {
+    QString saveFileName = QFileDialog::getSaveFileName(this, "Save File", m_fileName);
+    QFile file(saveFileName);
+    file.open(QFile::WriteOnly);
+    file.write(m_codeEdit->toPlainText().toLocal8Bit());
+    file.close();
+}
+
 void FileEditorMdiSubWindow::construct() {
     QStyle *style = QApplication::style();
     setWidget(new QWidget());
     m_toolBar = new QToolBar(this);
     m_codeEdit = new CodeEdit(this);
     m_statusBar = new QStatusBar(this);
 
     m_codeEdit->setFontFamily("Courier");
@@ -45,9 +55,10 @@ void FileEditorMdiSubWindow::construct()
     layout->addWidget(m_toolBar);
     layout->addWidget(m_codeEdit);
     layout->addWidget(m_statusBar);
     layout->setMargin(2);
     widget()->setLayout(layout);
 
     connect(undoAction, SIGNAL(triggered()), m_codeEdit, SLOT(undo()));
     connect(redoAction, SIGNAL(triggered()), m_codeEdit, SLOT(redo()));
+    connect(saveAction, SIGNAL(triggered()), this, SLOT(saveFile()));
 }
diff --git a/gui//src/FileEditorMdiSubWindow.h b/gui//src/FileEditorMdiSubWindow.h
--- a/gui//src/FileEditorMdiSubWindow.h
+++ b/gui//src/FileEditorMdiSubWindow.h
@@ -2,20 +2,25 @@
 #define FILEEDITORMDISUBWINDOW_H
 
 #include <QMdiSubWindow>
 #include <QToolBar>
 #include <QStatusBar>
 #include "CodeEdit.h"
 
 class FileEditorMdiSubWindow : public QMdiSubWindow {
+    Q_OBJECT
 public:
     FileEditorMdiSubWindow(QWidget *parent = 0);
     void loadFile(QString fileName);
 
+public slots:
+    void saveFile();
+
 private:
     void construct();
     QToolBar *m_toolBar;
     CodeEdit *m_codeEdit;
     QStatusBar *m_statusBar;
+    QString m_fileName;
 };
 
 #endif // FILEEDITORMDISUBWINDOW_H
